### PR TITLE
{chem,numlib}[GCC/11.3.0,gompi/2022a] GAMESS-US v20230930-R2 and v20220930-R2, OpenBLAS v0.3.20 w/ int8

### PR DESCRIPTION
**Pull request by lexming**
(created using `eb --new-pr`)

* [x] Depends on https://github.com/easybuilders/easybuild-easyblocks/pull/3047/

Build with GNU compilers and OpenBLAS. Uses `mpi` as comm target with OpenMPI for parallelism.

### CODE REVIEW

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
node303.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/28f8c5cea9b7760ed46ddaba1c5714f9 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
node303.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/7f330dc7d1d0abed3df4f61004c3b524 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in total)
node302.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/0d21fac0d69a594be3ac31b7b5c63917 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**FAILED**
Build succeeded for 2 out of 3 (3 easyconfigs in total)
node201.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz (broadwell), Python 3.6.8
See https://gist.github.com/lexming/3ab89a4c57442edb8e1d753482739998 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node315.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/ffd29e326d04903409f4729a9c3db291 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node201.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz (broadwell), Python 3.6.8
See https://gist.github.com/lexming/c6857c5c6dab475c8aa30d2a87828dd3 for a full test report.

**Comment by sassy-crick** [comment]
There are a few issues with the versioning. There are only the versions `GAMESS VERSION = 30 JUN 2023 (R1)` and `GAMESS VERSION = 30 SEP 2023 (R2)`. I also think you mixed up the original tarballs. I happen to have `gamess-30062023-R1.tar.gz` with sha256sum of `b52fc872669a14f0f12fc17ae50d313a046c778fc25d3307720745f91e6f397c` and `gamess-30092023-R2.tar.gz` with sha256sum `2b7cf4af17fb2eab5bf3609bf820437728cd36d87f44857dce25bafa9e9622ad`. 
Could you please double check what you have? You can find the correct version of the tarball you were using in the `gamess/source` directory like this:
```
grep "GAMESS VERSION" *
ecp.src:C        -----  GAMESS VERSION OCTOBER 1989.                     -----
fmolib.src: 9000 format(1x,'GAMESS VERSION')
gamess.src:C     ----- PRINT THE GAMESS VERSION BANNER -----
gamess.src:     *     10X,1H*,10X,'GAMESS VERSION = 30 JUN 2023 (R1)',
```
Or you simply look at the `gamess.src` file of course. 
Either way, please correct that so I could do the testing and upload a test report. 
On that note: I had problem with the `OpenBLAS-i8` build as quite some `LAPACK` tests failed on my laptop (11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz), more than the allowed ones. I will test on a few other machines to see if that is my CPU or a different issue.

**Comment by sassy-crick** [comment]
As mentioned, I tested `OpenBLAS` on a few machines: my laptop, our AMD Zen2 and the Intel Platinum 8358 we have. Only in case of the Intel Platinum 8358 all tests passed within the required limits. The test reports, and the log-files of the failed jobs, can be found [here](https://gist.github.com/sassy-crick/4815315476cb6aa08a73f91fd80be9a2). Happy to test more if required. 
I remember about beginning this year I was looking into an `OpenBLAS-i8` installation as well but I was running into issues, so I ditched that in favour of `MKL` for the `GAMESS-US` EasyConfigs I done, again some time ago.

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20220930-R2-gompi-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# rename gamess-current.tar.gz by changing 'current' to the proper version
+sources = ['gamess-%(version)s.tar.gz']
+checksums = ['22c8c2dc3ef007c36a32d9ff5a9e1803ac9006f24065455c22de40dd26fb2a7a']
+
+# custom location of gamess workdir
+# scratch_dir = '/path/to/fast/storage'
+# user_scratch_dir = '/path/to/fast/storage'
+
+maxcpus = '256'
+maxnodes = '32'
```
Comment (boegel) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20220930-R2-gompi-2022a.eb:37]: Maybe increase those a bit to something that's less likely to be a problem in practice.

Would also be good to reflect these maxima in the module description?

**Comment by lexming** [comment]
@sassy-crick sorry for the confusion, there was a typo in the version string of v20230930-R2. Fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/19310/commits/c5fe35bc3b63b54a9f7615f124a7337016c388a8

**Comment by sassy-crick** [comment]
@lexming Sorry for my confusion as well. I did not realise that one was from 2022. 

What would be more interesting is to see why the `OpenBLAS` installation is failing on the AMD Rome chip. I know that for my laptop I got some issues with that due to the CPU detection.

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20220930-R2-gompi-2022a.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20220930-R2-gompi-2022a.eb]: sure, fixed in https://github.com/easybuilders/easybuild-easyconfigs/pull/19310/commits/cd1391286bc0a231bbf240bf691287fb31fc44fd

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node302.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/8c6a821732dccd3c4d42198fc8bb53f0 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node200.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz (broadwell), Python 3.6.8
See https://gist.github.com/lexming/2b1e1250f16afd92628bbe8ba6303446 for a full test report.

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20220930-R2-gompi-2022a.eb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+toolchain = {'name': 'gompi', 'version': '2022a'}
+toolchainopts = {'usempi': True, 'openmp': False}
+
+dependencies = [
+    ('OpenBLAS', '0.3.20', '-int8'),
+    ('libxc', '5.2.3'),
+]
+
+# manually download via http://www.msg.chem.iastate.edu/gamess/download.html (requires registration)
+# rename gamess-current.tar.gz by changing 'current' to the proper version
```
Comment (jfgrimm) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20220930-R2-gompi-2022a.eb:34]: use `download_instructions` for this

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20230930-R2-gompi-2022a.eb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+toolchain = {'name': 'gompi', 'version': '2022a'}
+toolchainopts = {'usempi': True, 'openmp': False}
+
+dependencies = [
+    ('OpenBLAS', '0.3.20', '-int8'),
+    ('libxc', '5.2.3'),
+]
+
+# manually download via http://www.msg.chem.iastate.edu/gamess/download.html (requires registration)
+# rename gamess-current.tar.gz by changing 'current' to the proper version
```
Comment (jfgrimm) [inline easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20230930-R2-gompi-2022a.eb:34]: use `download_instructions` for this

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20220930-R2-gompi-2022a.eb (hunk 2 above)
Comment (lexming) [reply easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20220930-R2-gompi-2022a.eb]: good point, done in https://github.com/easybuilders/easybuild-easyconfigs/pull/19310/commits/20a852fb3e621c157ff6fbb35686a835fa563dec

File: easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20230930-R2-gompi-2022a.eb (hunk 3 above)
Comment (lexming) [reply easybuild/easyconfigs/g/GAMESS-US/GAMESS-US-20230930-R2-gompi-2022a.eb]: good point, done in https://github.com/easybuilders/easybuild-easyconfigs/pull/19310/commits/20a852fb3e621c157ff6fbb35686a835fa563dec

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node201.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz (broadwell), Python 3.6.8
See https://gist.github.com/lexming/498dd1903bb094a1f657ccef01106345 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node309.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/52cbc01a6b5b9611a8eb2b844667a7a2 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node200.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz (broadwell), Python 3.6.8
See https://gist.github.com/lexming/874130d170461b41e8b41c1b59fbaac4 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node300.hydra.os - Linux Rocky Linux 8.8, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/75ca45d6cbf3a9ffd09e419730e1f6ef for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
node300.hydra.os - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/c032347d28d6822cd968f30bee77b606 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/3ddc5650413cb6917cf73e6c993f31c6 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3047
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/963a62ce1c2073ef4502734a068e8b69 for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Going in, thanks @lexming!

**Pull request merged**